lightning/src/ln/channel.rs

```diff
@@ -4784,9 +4791,14 @@ impl<SP: Deref> Channel<SP> where
 						// not fail - any in between attempts to claim the HTLC will have resulted
 						// in it hitting the holding cell again and we cannot change the state of a
 						// holding cell HTLC from fulfill to anything else.
+						//
+						// Note that we should have already provided a preimage-containing
+						// `ChannelMonitorUpdate` to the user, making this one redundant, however
```

Comment:
I don't think I realized we could have a duplicate monitor update here. It's pre-existing, but may be useful to note on `ChannelMonitor::provide_payment_preimage` that preimages may be provided multiple times and it needs to be idempotent, since it's not obvious reading the method that it is. 

Also, is this duplicate update worth addressing in the future?